Sources/Web3Auth/Web3Auth.swift

```diff
@@ -30,16 +39,24 @@ public class Web3Auth: NSObject {
      */
     public init(_ params: W3AInitParams) async {
         initParams = params
+        Router.baseURL = SIGNER_MAP[params.network] ?? ""
         sessionManager = .init()
-            do {
-                let loginDetailsDict = try await sessionManager.authorizeSession()
-                guard let loginDetails = Web3AuthState(dict: loginDetailsDict, sessionID: sessionManager.getSessionID() ?? "",
-                network: initParams.network) else { throw Web3AuthError.decodingError }
-                state = loginDetails
-            } catch let error {
-                os_log("%s", log: getTorusLogger(log: Web3AuthLogger.core, type: .error), type: .error, error.localizedDescription)
-            }
         super.init()
+        do {
+            let fetchConfigResult = try await fetchProjectConfig()
+            if(fetchConfigResult) {
+                do {
+                    let loginDetailsDict = try await sessionManager.authorizeSession()
+                    guard let loginDetails = Web3AuthState(dict: loginDetailsDict, sessionID: sessionManager.getSessionID() ?? "",
+                    network: initParams.network) else { throw Web3AuthError.decodingError }
+                    state = loginDetails
+                } catch let error {
+                    os_log("%s", log: getTorusLogger(log: Web3AuthLogger.core, type: .error), type: .error, error.localizedDescription)
+                }
+            }
+        } catch let error {
+            os_log("%s", log: getTorusLogger(log: Web3AuthLogger.core, type: .error), type: .error, error.localizedDescription)
```

Comment:
init should fail if fetch config api fails

Reply:
pushed.